protocols/index/test/Index-unit.js

```diff
@@ -49,8 +56,9 @@ contract('Index Unit Tests', async accounts => {
       let listLength = await index.length()
       equal(listLength, 0, 'Link list length should be 0')
 
-      let locators = await index.fetchLocators(10)
-      equal(locators.length, 0, 'list should have 0 locators')
+      let locators = await index.fetchLocators(EMPTY_ADDRESS, 10)
+      equal(locators.length, 10, 'list should have 10 slots')
```

Comment:
10 slots but they're empty?
I guess this is a limitation with being unable to return dynamic arrays. This is odd from a user standpoint. I now have to check the bounds of my returned data because the length can be incorrect. I guess this was how Index was working before. Apologies for not catching this earlier when we were writing Index.

Reply:
To be clear this is a non issue. This is just a comment.

Reply:
To be clear this is just a comment.

Reply:
This was not how the index was working before. Previously you could check which was smaller of `_count` and `length` and create an array of that size before filling it up. This is no longer possible as you aren't starting from the beginning of the list and so you cannot guarantee how long it will be beforehand.

I spoke to Don about it to see if it was an issue because the alternative is creating a new array at the end of the function and copying across the locators one by one, and we concluded this wasnt worth it. Realistically it'll just be used by front ends so they can check

Reply:
> I guess this was how Index was working before. Apologies for not catching this earlier when we were writing Index.

I don't know why I said this. I swore I saw something earlier - tired eyes.